arrow-libs/optics/arrow-optics/src/main/kotlin/arrow/optics/Fold.kt

```diff
@@ -35,178 +16,154 @@ inline fun <S, A> FoldOf<S, A>.fix(): Fold<S, A> = this as Fold<S, A>
  * @param S the source of a [Fold]
  * @param A the target of a [Fold]
  */
-interface Fold<S, A> : FoldOf<S, A> {
+interface Fold<S, A> {
 
   /**
    * Map each target to a type R and use a Monoid to fold the results
    */
-  fun <R> foldMap(M: Monoid<R>, s: S, f: (A) -> R): R
-
-  companion object {
-
-    fun <A> id() = PIso.id<A>().asFold()
-
-    /**
-     * [Fold] that takes either [S] or [S] and strips the choice of [S].
-     */
-    fun <S> codiagonal() = object : Fold<Either<S, S>, S> {
-      override fun <R> foldMap(M: Monoid<R>, s: Either<S, S>, f: (S) -> R): R = s.fold(f, f)
-    }
-
-    /**
-     * Creates a [Fold] based on a predicate of the source [S]
-     */
-    fun <S> select(p: (S) -> Boolean): Fold<S, S> = object : Fold<S, S> {
-      override fun <R> foldMap(M: Monoid<R>, s: S, f: (S) -> R): R = if (p(s)) f(s) else M.empty()
-    }
-
-    /**
-     * [Fold] that points to nothing
-     */
-    fun <A, B> void() = POptional.void<A, B>().asFold()
-
-    /**
-     * Create a [Fold] from a [arrow.Foldable]
-     */
-    fun <F, S> fromFoldable(foldable: Foldable<F>) = object : Fold<Kind<F, S>, S> {
-      override fun <R> foldMap(M: Monoid<R>, s: Kind<F, S>, f: (S) -> R): R = foldable.run { s.foldMap(M, f) }
-    }
-  }
+  fun <R> foldMap(M: Monoid<R>, source: S, map: (focus: A) -> R): R
 
   /**
    * Calculate the number of targets
    */
-  fun size(s: S) = foldMap(Int.monoid(), s = s, f = { _ -> 1 })
+  fun size(source: S) =
+    foldMap(Monoid.int(), source) { 1 }
 
   /**
    * Check if all targets satisfy the predicate
    */
-  fun forall(s: S, p: (A) -> Boolean): Boolean = foldMap(AndMonoid, s, p)
+  fun all(source: S, predicate: (focus: A) -> Boolean): Boolean =
+    foldMap(Monoid.boolean(), source, predicate)
+
+  fun forall(source: S, predicate: (focus: A) -> Boolean): Boolean =
```

Comment:
If I recall well, we deprecated `forall` methods in favor of `all` to follow the Kotlin Standard naming, didn't we? Should we do it here too?

Reply:
Yes, absolutely! Good catch